src/Results/DTO/GenerativeAiResult.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Results\DTO;
+
+use WordPress\AiClient\Files\Contracts\FileInterface;
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\Enums\MessagePartTypeEnum;
+use WordPress\AiClient\Results\Contracts\ResultInterface;
+
+/**
+ * Represents the result of a generative AI operation.
+ *
+ * This DTO contains the generated candidates along with usage statistics
+ * and metadata from the AI provider.
+ *
+ * @since n.e.x.t
+ */
+class GenerativeAiResult implements ResultInterface
+{
+    /**
+     * @var string Unique identifier for this result.
+     */
+    private string $id;
+
+    /**
+     * @var Candidate[] The generated candidates.
+     */
+    private array $candidates;
+
+    /**
+     * @var TokenUsage Token usage statistics.
+     */
+    private TokenUsage $tokenUsage;
+
+    /**
+     * @var array<string, mixed> Provider-specific metadata.
+     */
+    private array $providerMetadata;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $id Unique identifier for this result.
+     * @param Candidate[] $candidates The generated candidates.
+     * @param TokenUsage $tokenUsage Token usage statistics.
+     * @param array<string, mixed> $providerMetadata Provider-specific metadata.
+     */
+    public function __construct(string $id, array $candidates, TokenUsage $tokenUsage, array $providerMetadata = [])
+    {
+        $this->id = $id;
+        $this->candidates = $candidates;
```

Comment:
Instead of validating in all the individual output methods, shouldn't we validate in here? `$candidates` must always contain at least 1 entry.

Reply:
Resolved in d51a35f